src/main.qml

```diff
@@ -342,6 +342,20 @@ ApplicationWindow {
                     }
                 }
             }
+
+            DropArea {
+                anchors.fill: parent
+                onEntered: {
+                    if (drag.active && mimeData.hasUrls()) {
+                        drag.acceptProposedAction()
+                    }
+                }
+                onDropped: {
+                    if (drop.urls && drop.urls.length > 0) {
```

Comment:
Here you can use the [optional chaining operator](https://doc.qt.io/qt-6/qtqml-javascript-hostenvironment.html#common-base) to make the condition cleaner:
```qml
if (drop.urls?.length > 0) {
```

Reply:
Better not...
Not supported on Qt 5.